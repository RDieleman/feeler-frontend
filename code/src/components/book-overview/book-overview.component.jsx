import React from "react";
import "./book-overview.styles.css";

export const BookOverview = ({books, handleOnClick}) => {

    return (
        <div className="book-overview-container">
            {books.map((b, i) => {
                return <div
                    onClick={() => handleOnClick(b)}
                    key={i}
                >
                    <img src={b.imageUrl} alt={b.title}/>
                </div>
            })}
        </div>
    )
}