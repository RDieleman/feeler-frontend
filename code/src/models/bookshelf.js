//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn13,
                i.status
            )
        );
    })
    this.getRead = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READ") {
                items.push(i);
            }
        });
        return items;
    }

    this.getUnread = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "UNREAD") {
                items.push(i);
            }
        });
        return items;
    }

    this.getReading = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READING") {
                items.push(i);
            }
        });
        return items;
    }
}