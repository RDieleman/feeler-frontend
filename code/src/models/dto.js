export function AddBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function GetBookshelfDTO(bookshelfId){
    this.bookshelfId = bookshelfId;
}

export function RemoveBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function UpdateBookDTO(bookshelfId, isbn, status){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
    this.status = status;
}

