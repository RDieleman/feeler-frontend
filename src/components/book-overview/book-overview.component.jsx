import React from "react";
import "./book-overview.styles.css";

export const BookOverview = ({books}) => {

    return (
        <div className="book-overview-container">
            {books.map((b, i) => {
                return <div
                key={i}
                >
                    <img src={b.imageUrl} alt={b.title}/>
                </div>
            })}
        </div>
    )
}