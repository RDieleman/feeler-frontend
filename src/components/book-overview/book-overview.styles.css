.book-overview-container{
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
}

.book-overview-container > div{
    background-color: var(--color-gen-medium);
    border-radius: var(--border-radius);
    width: 100%;
    max-width: 450px;
}

.book-overview-container > div > img{
    width: 100%;
    height: auto;
}