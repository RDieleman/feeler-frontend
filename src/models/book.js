//Book info
export function Book(title, overview, isbn, imageUrl){
    this.title = title;
    this.overview = (overview) ? overview : "No description found.";
    this.isbn = isbn;
    this.imageUrl = imageUrl;
}