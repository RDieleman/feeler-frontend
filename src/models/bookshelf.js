//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn,
                i.status
            )
        );
    })
}