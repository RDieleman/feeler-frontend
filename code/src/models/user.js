import {Bookshelf} from "./bookshelf";

export function User(id, name, bookshelf){
    this.id = id;
    this.name = name;
    this.bookshelf = new Bookshelf(
        bookshelf.id,
        bookshelf.content
    );
}