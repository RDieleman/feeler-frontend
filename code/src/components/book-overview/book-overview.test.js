import React from "react";
import { shallow } from "enzyme";
import {BookOverview} from "./book-overview.component";
import {Book} from "../../models/book";

const books = [
    new Book(
        "test title 1",
        "test overview 1",
        "12345",
        "test image url 1"
    ),
    new Book(
        "test title 2",
        "test overview 2",
        "12345",
        "test image url 2"
    )
]

it("Expect to render Book Overview component", () =>{
    expect(shallow(
        <BookOverview
            handleOnClick={() => console.log("Book clicked")}
            books={books}
    />)).toMatchSnapshot();
});


