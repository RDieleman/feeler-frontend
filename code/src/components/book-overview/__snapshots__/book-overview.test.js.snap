// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Book Overview component 1`] = `ShallowWrapper {}`;
